mysql-test/lib/mtr_report.pm

```diff
@@ -499,27 +499,39 @@ sub mtr_report_stats ($$$$) {
 
       my $test_result;
 
-      # if a test case has to be retried it should have the result MTR_RES_FAILED in jUnit XML
       if ($test->{'retries'} > 0) {
         $test_result = "MTR_RES_FAILED";
+        if ($test->{'failures'} > 1){
+          $test_result = "MTR_RES_RETRY_FAILED";
+        }
+        if ($test->{'result'} eq "MTR_RES_PASSED"){
```

Comment:
Shouldn't here be 
```
if (failure)
{}
else
{}
```
Is it possible to have failure >1 and result different from MTR_RES_FAILED?

Reply:
Yes, it is possible. For each repeated test, `failure >=1`. If the test failed twice and then succeeded, then `failure = 2`, but the status will be `MTR_RES_PASSED` The structure with else is not suitable here.

Reply:
Irrelevant of the number of failures, result of failed test will always be `MTR_RES_FAILED`.
That's why you can check 
```
if ($test->{'result'} eq "MTR_RES_FAILED") 
{} 
else 
{}
```

Reply:
@timofeyturenko , @DmitriyKarpovskiy1  please use if-else statement suggested in the last comment.
We may have situation of having `pass -> fail ->  retry-pass -> retry-fail -> retry-pass`  depending of the number of attempts (`--retry`),  where for the last attempt, number of failures >1 but the test passed and  your CI should note that correctly.
Please squash to single commit.

Reply:
I don't understand why you need if {} else {}. Your version doesn't work. Any test that has `retry-failure > 1` and it has `failed test - >{'retries'} = retry-failure ( > 1)`. That is, this will cause every first test to fall to have the `MTR_RES_RETRY_FAILED` attribute, but in fact it should just have `RETRY_FAILED`

Reply:
See below, less assignment and clearer code with nested if conditions.